### PR TITLE
Fix high contrast focus state on vertical menu

### PR DESCRIPTION
**Pull request by TanelVari**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

VSO issue #701692

#### Description of changes

Keyboard navigation now has bounding rectangle. Background doesn't change anymore.

#### Focus areas to test

Vertical menus for Menu component

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12462)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 1]
```diff
@@ -29,21 +29,29 @@ const getFocusedStyles = ({
...
+      border: `solid 1px ${v.borderColorFocus}`,
+      outline: `solid 1px ${v.outlineColorFocus}`,
+      margin: pxToRem(1),
+      background: v.verticalBackgroundColorFocus,
+      color: v.colorFocus || colors.foregroundFocus,
+
+      ...(primary && { color: v.color }),
+
+      ...(active && {
+        color: primary ? colors.foregroundFocus : v.colorActive,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts]:
```suggestion
        color: v.colorActive,
        ...(primary && { color: colors.foregroundFocus }),
```

Let's avoid inline ternaries for props

File: packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 2]
```diff
@@ -29,21 +29,29 @@ const getFocusedStyles = ({
...
-  return {
-    color: v.colorActive,
-    background: v.backgroundColorFocus || colors.backgroundFocus,
 
-    ...(primary && {
-      color: colors.foregroundFocus,
+  return {
+    ...(!vertical && {
+      color: primary ? colors.foregroundFocus : v.colorActive || colors.foregroundActive,
+      background: primary ? colors.backgroundFocus : v.backgroundColorFocus || colors.backgroundFocus,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts]:
```suggestion
      color: v.colorActive || colors.foregroundActive,
      background: v.backgroundColorFocus || colors.backgroundFocus,
      
      ...(primary && {
       color: colors.foregroundFocus,
       background: colors.backgroundFocus,
      }),
```

File: packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 3]
```diff
@@ -29,21 +29,29 @@ const getFocusedStyles = ({
...
+
   if (active && !underlined && !vertical) return {};
-  return {
-    color: v.colorActive,
-    background: v.backgroundColorFocus || colors.backgroundFocus,
 
-    ...(primary && {
-      color: colors.foregroundFocus,
+  return {
+    ...(!vertical && {
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts]: Can we avoid there a condition for `!vertical`? I.e.

```
...vertical && {
  color: someValue,
  background: someValue,
}
```

File: packages/fluentui/CHANGELOG.md [hunk 4]
```diff
@@ -27,6 +27,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Add `inverted` prop to propTypes check in `Button` @silviuavram ([#12414](https://github.com/microsoft/fluentui/pull/12414))
 - Fix hover color for disabled `Dropdown` trigger button @silviuavram ([#12418](https://github.com/microsoft/fluentui/pull/12418))
 - Fix `ButtonContent` className and added conformat test @mnajdova ([#12431](https://github.com/microsoft/fluentui/pull/12431))
+- Fix keyboard navigation for vertical Menu components in Teams theme high contrast mode @TanelVari ([#12462](https://github.com/microsoft/fluentui/pull/12462))
```
Comment (pompomon) [inline packages/fluentui/CHANGELOG.md]: Please could you clarify the comment in Changelog, I don't see any changes related to keyboard navigation per say, only color changes

File: packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts (hunk 3 above)
Comment (TanelVari) [reply packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts:36]: Thanks

File: packages/fluentui/CHANGELOG.md (hunk 4 above)
Comment (TanelVari) [reply packages/fluentui/CHANGELOG.md]: Good point.

**Pull request merged**